hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/contracts/precompile/impl/TransferPrecompile.java

```diff
@@ -678,4 +688,138 @@ private AccountAmount aaWith(final AccountID account, final long amount, final b
     public void setCanFallbackToApprovals(final boolean canFallbackToApprovals) {
         this.canFallbackToApprovals = canFallbackToApprovals;
     }
+
+    /**
+     * Given a {@link BalanceChange} that was switched from key-based authorization to approval-based authorization,
+     * tries to update the matching adjustment in the transfer list of the in-progress synthetic {@code CryptoTransfer}
+     * op to reflect the switch.
+     *
+     * <p><b>Important:</b> this method acts via side effects, mutating the given {@code opBuilder} in-place.
+     *
+     * @param opBuilder the builder for the in-progress synthetic {@code CryptoTransfer} op
+     * @param switchedChange the {@code BalanceChange} switched to approval-based authorization
+     */
+    @VisibleForTesting
+    static void updateSynthOpForAutoApprovalOf(
+            @NonNull final CryptoTransferTransactionBody.Builder opBuilder,
+            @NonNull final BalanceChange switchedChange) {
```

Comment:
We should probably mention this switched change is only for debit in a comment ?

Reply:
Or better yet check in the code that it's a debit

Reply:
I think it will not reach to this function if its not debit 😃